Add bit manipulation methods

simo498 opened the pull request:
It adds some methods to check and set single bits. For now there aren't methods to handle multiple bits masks, and it's limited to u32 (due to the assence of a common Integer trait  in rust).

Refers to #33

simo498 commented:
> @simo498 here I appreciate more feedback also from @AlessandroGrassi99 and @gallottino 
> But sure I prefer this kind of api not for a struct `Bitwise` but as implementation of `u32` or such... In this way you can do `my_number.toggle_bit(15)` and this will toggle the bit 15.
> I have some doubt also if is better `is_set/is_reset` but let's discuss :D

Yeah it's an idea. Another is to create a wrapper like ’Bitwise(u32)’. Anyway let's hear the opinion of other people on which implementation we should uses. I also think something more direct like u32::get_bit or whatever would be more readable

gallottino commented on src/bitwise.rs:
```diff
@@ -0,0 +1,93 @@
+    use super::*;
+
+    #[test]
+    fn basic_tests() {
```
Here I prefer to split  the tests to recognize faster a future change that broke this implementation.  You can also take this test with all implementations, but I think  also it is better to have almost one test for every functionality.

gallottino reviewed (changes requested):
I also like that you create a struct from u32 and use it to manipulate the inside values. If you can change it will be great. u32 as parameter in every function is a bit redundant and also less readable. Be careful to test every function with at least an atomic test.

simo498 commented:
I implemented the methods to u32 directly, so they can be called by just adding `use crate::bitwise::Bits`, i also added `rand` for testing, but it can be useful in the future as well

guerinoni commented on Cargo.toml:
```diff
@@ -6,6 +6,7 @@ edition = "2021"
 [dependencies]
 egui_glium = "0.19.0"
 glium = "0.32"
+rand = "0.8.5"
```
Can you move this as dev-dependencies?

simo498 replied:
Sure

guerinoni reviewed (changes requested):
Looks nicer :)

guerinoni reviewed (approved):
Good :)
